### ISSUE TITLE
Inconsistent menu font size when dragging mail to a QuickFolders tab

### ISSUE THREAD

**Issue by FTKL** [RealRaven2000/QuickFolders#329]
This problem is only visible when the button font size option under "Personalize my tabs!" has been changed to something other than the default value.
The button font size (extensions.quickfolders.buttonFontSizeN) also changes accordingly the list font size when right-clicking on the button. However, when dragging an email to the button, the list font sizes remains the default one.

Steps to replicate : change the the button font size. Drag an email to a pinned folder. Notice the difference compared to right-clicking the pinned folder.

**Comment by RealRaven2000** [comment]
will need to investigate - sorry for the late reply

**Comment by RealRaven2000** [comment]
in my experience, the font size of the dropdown menus is never affected by the button font size - whether you drag emails to the tab or right-click it:

![image](https://user-images.githubusercontent.com/6314674/208672429-2d1f048e-3c33-4934-b0bf-0c2bbbbfbfc8.png)

**Comment by FTKL** [comment]
Here is what I see with Button font size set on 10.
When I change this value:
[image: image.png]

Tab font size shrinks accordingly:
[image: image.png]
When I right-cilck to have the dropdown, it stays the same
[image: image.png]
When I drag and drop a mail over it, the text inside the dropdown reverts
to default while text of the tab remains the same
[image: image.png]

Best,
Raphaël




Le mar. 20 déc. 2022 à 13:56, RealRaven2000 ***@***.***> a
écrit :

**Comment by FTKL** [comment]
done

**Comment by RealRaven2000** [comment]
I could reproduce the behavior now - I had tested in a test profile on Thunderbird 91 where the font size wasn't affected when you right-clicked or dragged. Indeed it is inconsistent in Thunderbird 102. (By the way it is easier to see if you set a large font size). It's a strange one as both menus are built with the same function. Maybe they are "attached" to the tabs in different ways.

**Comment by RealRaven2000** [comment]
When debugging and having a breakpoint during the drag and drop I see that the popup menu is not appended within the button element. It is appended after the last visible tab.

When right-clicking a button, the popup menu is a child of the `toolbarbutton` element
![image](https://user-images.githubusercontent.com/6314674/208904850-6ffc74e5-2dbb-46bc-ae64-7f79ec74d594.png)

In inspector, I tried adding a rule to override `font-size` and a `color` attribute for testing:

```css
#QuickFolders-FoldersBox popupset label {
  font-size: 14px !important;
  color: red;
}
```
![image](https://user-images.githubusercontent.com/6314674/208913609-c6caa141-4259-4d5f-9c56-c53542898a4c.png)

this looks promising, I will add this to the code of the updateUserStyles() function that is in charge of updating the layout (blingable) elements of the toolbar. Probably need a second rule for the navigation bar (Current folder toolbar) for consistency.

**Comment by RealRaven2000** [comment]
Here is a preview version for testing that implements the new rule to fix this inconsistency.

[QuickFolders-mx-5.14.1pre24.zip](https://github.com/RealRaven2000/QuickFolders/files/10278080/QuickFolders-mx-5.14.1pre24.zip)

I also decided to apply the font size to the results of the quickMove search results (but not to the textbox), for consistency:

![image](https://user-images.githubusercontent.com/6314674/208918555-4c6e0777-3a6a-4f5e-972a-08b1823d2772.png)

To test this version, download the zip file above and drag it into Thunderbird Add-ons Manager (without extracting its contents)

**Comment by RealRaven2000** [comment]
There was still an inconsistency in Thunderbird 91 (when right-clicking a QF tab, the custom font size wouldn't work) - so I added an additional specific rule for Thunderbird 91. Prerelease version below:

[QuickFolders-mx-5.15pre31.zip](https://github.com/RealRaven2000/QuickFolders/files/10279239/QuickFolders-mx-5.15pre31.zip)

To test this version, download the zip file above and drag it into Thunderbird Add-ons Manager (without extracting its contents)

**Comment by RealRaven2000** [comment]
Harmonized rules so same rule can be applied to 91 and 102:

[QuickFolders-mx-5.15pre33.zip](https://github.com/RealRaven2000/QuickFolders/files/10279297/QuickFolders-mx-5.15pre33.zip)

To test this version, download the zip file above and drag it into Thunderbird Add-ons Manager (without extracting its contents)

**Comment by RealRaven2000** [comment]
Implemented in 5.15.1 - Published 22/Dec/2022
